onboarding/mlsys/1-backprop.md

```diff
@@ -0,0 +1,414 @@
+# Part 1: Backpropagation
+
+Machine learning is all about getting computers to solve math problems for you.
+
+Specifically, it's about solving math problems like, "For a given mathematical
+model, what weights in the model would cause the model to calculate the most
+accurate predictions for a set of inputs?"
+
+You're going to write code to solve this exact problem for one particularly
+simple model.
+
+Afterwards, you will understand of basic concepts like gradient descent and
+batch size.  You will also understand how those basic concepts translate into
+some of Determined's advanced features, like gradient aggregation,
+data-parallel distributed training, and model-parallel distributed training.
+
+## The Model
+
+Our model is going to be be a simple linear model: `ypred = m*x` (`pred` is
+short for "predicted"). In statistics they would say there is no bias term, or
+in algebra they would there's no constant term.  Just a stright line through
+the origin.
+
+## The Data Set
+
+Our made-up dataset has 4 data points:
+
+- `(x=1, ylabel=1)`
+- `(x=2, ylabel=2)`
+- `(x=3, ylabel=3)`
+- `(x=4, ylabel=4)`
+
+## The Math Problem
+
+What is the best value of `m` to make our model (`ypred = m*x`) most-accurately
+predict our dataset?
+
+### Q1:
+
+The answer is `m=1`.  Looking at the dataset, explain why `m=1` is correct
+before proceeding with further math.
+
+## What does "most-accurately" mean, really?
+
+For this model and dataset, there is an exact answer (`m`=1), but usually there
```

Comment:
```suggestion
For this model and dataset, there is an exact answer (`m=1`), but usually there
```

And perhaps "perfect" in place of "exact"? There's always some exact best answer; it just won't have all the residuals be 0.